### PR TITLE
Remove redundant shell definition

### PR DESCRIPTION
**Pull request by rdhar**
Remove `shell: bash` from all jobs' steps since `defaults: run: shell: bash` has already been defined at the [top of the file](https://github.com/hashicorp/setup-terraform/compare/main...rdhar:setup-terraform:patch-1#diff-1c00228f84c2234bd6f3fd35eb6745ca9effb272abb50d7a48dcc7a8b4fd64e0L10-R11), which applies to all jobs in the workflow [per documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaultsrun).

### CODE REVIEW

File: .github/workflows/setup-terraform.yml
Comment (austinvalle) [inline .github/workflows/setup-terraform.yml:1]: GitHub isn't letting me make a comment suggestion on the exact lines, so just leaving it on the file itself 😄 

The top level default you linked is being overridden [here](https://github.com/hashicorp/setup-terraform/pull/358/files#diff-1c00228f84c2234bd6f3fd35eb6745ca9effb272abb50d7a48dcc7a8b4fd64e0R254-R256) and [here](https://github.com/hashicorp/setup-terraform/pull/358/files#diff-1c00228f84c2234bd6f3fd35eb6745ca9effb272abb50d7a48dcc7a8b4fd64e0R283-R285). If you'd like to update your PR to include bash as a default in these two setting blocks then I'm good with the changes 👍🏻 

```yaml
    defaults:
      run:
        shell: bash
        working-directory: ./.github/workflows/data/local
```

You can see the shell is ignoring the default here: https://github.com/hashicorp/setup-terraform/actions/runs/6546998008/job/17778750463?pr=358#step:5:3

**Review by austinvalle — CHANGES_REQUESTED** [review]
Thanks for the contribution @rdhar! I have one note to address before merging 👍🏻

File: .github/workflows/setup-terraform.yml
Comment (rdhar) [reply .github/workflows/setup-terraform.yml:1]: Ah, I hadn't realized that re-defining `defaults: run: working-directory:` at a job-level would also disregard `defaults: run: shell:` set at the top-level... TIL!

**Review by austinvalle — APPROVED** [review]
lgtm, thanks again for the PR 🚀

**Pull request merged**